tests/rptest/tests/partition_movement_upgrade_test.py

```diff
@@ -63,11 +63,17 @@ def _start_consumer(self, topic_name):
 
     def verify(self):
         self.producer.wait()
+
+        def finished_consuming():
+            self.logger.debug(
+                f"verifying, producer acked: {self.producer.produce_status.acked}, "
+                f"consumer valid reads: {self.consumer.consumer_status.validator.valid_reads}"
+            )
+            return self.consumer.consumer_status.validator.valid_reads >= self.producer.produce_status.acked
+
         # wait for consumers to finish
-        wait_until(
-            lambda: self.consumer.consumer_status.valid_reads == self.producer.
-            produce_status.acked, 90)
-        self.consumer.shutdown()
+        wait_until(finished_consuming, 90)
```

Comment:
same as with the si leak test - should we cherry-pick https://github.com/redpanda-data/redpanda/pull/5862 first? It was even marked for cherry-picking